os/arch/arm/src/armv8-m/up_assertlogs.c

```diff
@@ -0,0 +1,570 @@
+#undef  CONFIG_DEBUG
+#undef  CONFIG_DEBUG_ERROR
+#undef  CONFIG_DEBUG_WARN
+#undef  CONFIG_DEBUG_VERBOSE
+#undef  CONFIG_LOGM
+#define CONFIG_DEBUG 1
+#define CONFIG_DEBUG_ERROR 1
+#define CONFIG_DEBUG_WARN 1
+#define CONFIG_DEBUG_VERBOSE 1
+#endif
+
+#include <sys/types.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <debug.h>
+
+#include <tinyara/irq.h>
+#include <tinyara/arch.h>
+#include <tinyara/board.h>
+#include <tinyara/syslog/syslog.h>
+
+#include <arch/board/board.h>
+#include <tinyara/sched.h>
+
+#include <tinyara/mm/mm.h>
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+#include <tinyara/binfmt/elf.h>
+#endif
+
+#ifdef CONFIG_SYSTEM_REBOOT_REASON
+#include <arch/reboot_reason.h>
+#endif
+#include "sched/sched.h"
+#ifdef CONFIG_BOARD_ASSERT_AUTORESET
+#include <sys/boardctl.h>
+#endif
+#ifdef CONFIG_BINMGR_RECOVERY
+#include <stdbool.h>
+#include <unistd.h>
+#include <queue.h>
+#include <tinyara/wdog.h>
+#include "semaphore/semaphore.h"
+#include "binary_manager/binary_manager.h"
+#endif
+#if defined(CONFIG_DEBUG_WORKQUEUE) && defined(__KERNEL__)
+#include <tinyara/wqueue.h>
+#endif
+#include "irq/irq.h"
+
+#include "up_arch.h"
+#include "up_internal.h"
+#include "mpu.h"
+
+bool abort_mode = false;
+
+#ifdef CONFIG_BINMGR_RECOVERY
+extern struct tcb_s *g_faultmsg_sender;
+extern sq_queue_t g_faultmsg_list;
+extern sq_queue_t g_freemsg_list;
+#endif
+
+extern uint32_t system_exception_location;
+extern uint32_t user_assert_location;
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+/* USB trace dumping */
+
+#ifndef CONFIG_USBDEV_TRACE
+#undef CONFIG_ARCH_USBDUMP
+#endif
+
+#ifndef CONFIG_BOARD_RESET_ON_ASSERT
+#define CONFIG_BOARD_RESET_ON_ASSERT 0
+#endif
+
+#define IS_FAULT_IN_USER_THREAD(fault_tcb)  ((void *)fault_tcb->uheap != NULL)
+#define IS_FAULT_IN_USER_SPACE(asserted_location)   (is_kernel_space((void *)asserted_location) == false)
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_stackdump
+ ****************************************************************************/
+
+#ifdef CONFIG_ARCH_STACKDUMP
+static void up_stackdump(uint32_t sp, uint32_t stack_base)
+{
+	uint32_t stack = sp & ~0x1f;
+	uint32_t *ptr = (uint32_t *)stack;
+	uint8_t i;
+
+	lldbg("%08x:", stack);
+	for (i = 0; i < 8; i++) {  // Print first 8 values for sp located
+		if (stack < sp) {
+			/* If stack pointer(sp) is aligned(sp & 0x1f) to an address outside allocated stack */
```

Comment:
(Minor) could you use a multi line comment since these 2 comment lines are the same comment in continuation?